tests/test_xyz/test_points.py

```diff
@@ -0,0 +1,135 @@
+import itertools
+import pathlib
+from hypothesis import given, strategies as st
+
+import pandas as pd
+import numpy as np
+import pytest
+
+from xtgeo.xyz import Points
+import tests.test_common.test_xtg as tsetup
+
+PFILE = pathlib.Path("points/eme/1/emerald_10_random.poi")
+POINTSET2 = pathlib.Path("points/reek/1/pointset2.poi")
+POINTSET3 = pathlib.Path("points/battle/1/many.rmsattr")
+POINTSET4 = pathlib.Path("points/reek/1/poi_attr.rmsattr")
+CSV1 = pathlib.Path("3dgrids/etc/gridqc1_rms_cellcenter.csv")
+
+
+def test_custom_points():
+    """Make points from list of tuples."""
+
+    plist = [(234, 556, 11), (235, 559, 14), (255, 577, 12)]
+
+    mypoints = Points(plist)
+
+    x0 = mypoints.dataframe["X_UTME"].values[0]
+    z2 = mypoints.dataframe["Z_TVDSS"].values[2]
+    assert x0 == 234
+    assert z2 == 12
+
+
+@st.composite
+def list_of_equal_length_lists(draw):
+    list_len = draw(st.integers(min_value=3, max_value=4))
+    fixed_len_list = st.lists(
+        st.floats(allow_nan=False, allow_infinity=False),
+        min_size=list_len,
+        max_size=list_len,
+    )
+    return draw(st.lists(fixed_len_list, min_size=1))
+
+
+@given(list_of_equal_length_lists())
+def test_create_pointset(points):
+    pointset = Points(points)
```

Comment:
docstring

Reply:
Will add.

I have to make a small comment though, that typically the naming of the test should be significant enough to describe what the test does (i.e. we should focus on good names for test function).

If it requires a docstring we're likely testing more than we should, with some exceptions.

I am not in favour of superfluous docstring (e.g. copy paste function name), and putting less information into the test name in order for the docstring to have a point of existence is also a poor choice.

I agree though that we should keep within the style of the code base, so I'll fix all the docstrings :)

Reply:
Good point. Will consider to make exception for test code.